app/src/main/java/com/github/se/signify/ui/Utils.kt

```diff
@@ -454,65 +454,72 @@ fun InfoPopup(onDismiss: () -> Unit, helpTitle: String, helpText: String) {
-      horizontalArrangement = Arrangement.SpaceBetween,
-      verticalAlignment = Alignment.CenterVertically) {
+  // Ensure that the lists have the same size
+  require(statsTextList.size == statsNumberList.size) { "The lists must have the same size." }
+  // Construction of the statistic column
+  Column(
+      modifier = Modifier.fillMaxWidth().testTag(columnTestTag),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Center) {
         Text(
             text = lineText,
             fontSize = 16.sp,
             color = MaterialTheme.colorScheme.onBackground,
             modifier = Modifier.testTag(lineTextTag))
-        for (index in columnTextList.indices) {
-          Column(
-              modifier =
-                  Modifier.size(50.dp)
-                      .border(
-                          2.dp, MaterialTheme.colorScheme.outlineVariant, RoundedCornerShape(12.dp))
-                      .clip(RoundedCornerShape(12.dp))
-                      .background(MaterialTheme.colorScheme.background)
-                      .testTag(columnTextTagList[index]),
-              horizontalAlignment = Alignment.CenterHorizontally,
-              verticalArrangement = Arrangement.Center) {
-                for (subIndex in columnTextList[index].indices) {
-                  Text(
-                      text = columnTextList[index][subIndex],
-                      fontSize = columnTextSPList[index][subIndex].sp,
-                      color = MaterialTheme.colorScheme.onBackground,
-                      modifier = Modifier.testTag(columnTextList[index][subIndex]))
-                }
+        Spacer(modifier = Modifier.height(12.dp))
+        Row(
+            modifier = Modifier.fillMaxWidth().testTag(rowTestTag),
+            horizontalArrangement = Arrangement.SpaceEvenly,
+            verticalAlignment = Alignment.CenterVertically) {
+              for (index in statsTextList.indices) {
+                Row(
+                    horizontalArrangement = Arrangement.SpaceBetween,
+                    verticalAlignment = Alignment.CenterVertically) {
+                      Text(
+                          text = statsTextList[index],
+                          fontSize = 12.sp,
+                          color = MaterialTheme.colorScheme.onBackground,
+                          modifier = Modifier.testTag(statsTextList[index]))
+                      Spacer(modifier = Modifier.width(12.dp))
+                      Row(
+                          modifier =
+                              Modifier.size(40.dp)
+                                  .border(
+                                      2.dp,
+                                      MaterialTheme.colorScheme.outlineVariant,
+                                      RoundedCornerShape(12.dp))
+                                  .clip(RoundedCornerShape(12.dp))
+                                  .background(MaterialTheme.colorScheme.background),
```

Comment:
This is way too much indentation. I have no clue where we are in the `@Composable`. You can use helper functions to factor sub-elements out.

Reply:
Noted